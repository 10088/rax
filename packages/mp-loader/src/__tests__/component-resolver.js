const { join } = require('path');
const { resolveComponentPath } = require('../component-resolver');

const projectPath = join(__dirname, '__files__/test-project');
const pagePath = join(projectPath, 'pages/foo/foo.js');
const componentPath = join(projectPath, '/components/foo/foo');
const subComponentPath = join(pagePath, '/sub-component/index');

describe('component resolver', () => {
  const r = resolveComponentPath;
  it('should resolve component path', () => {
    expect(
      r('/components/foo/foo', projectPath, pagePath)
    ).toEqual(componentPath);

    expect(
      r('../../components/foo/foo', projectPath, pagePath)
    ).toEqual(componentPath);

    expect(
      r('./sub-component', projectPath, pagePath)
    ).toEqual(subComponentPath);
  });

  // todo: npm add to git
  it('should resolve npm component', () => {
    expect(
      r('npm-pkg', projectPath, pagePath)
    ).toEqual(componentPath);
  });
});
