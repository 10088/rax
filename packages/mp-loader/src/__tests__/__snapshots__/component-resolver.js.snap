// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component resolver should resolve config properly 1`] = `
Object {
  "absolute": "/Users/zhuoling/workspace/rax/packages/mp-loader/src/__tests__/test-project/components/foo/index",
  "npm": "/Users/zhuoling/workspace/rax/packages/mp-loader/src/__tests__/test-project/node_modules/npm-pkg/es/card/index",
  "plugin": "plugin://foo-plugin/foo-component",
  "relative": "/Users/zhuoling/workspace/rax/packages/mp-loader/src/__tests__/test-project/components/foo/index",
  "self": "__SELF__",
}
`;
